docs/surveys/2019-10_3-months-of-govuk-docker.md

```diff
@@ -0,0 +1,291 @@
+have experienced, reasons why developers aren't using the tool, the wants
+for data replication and impressions on ease-of-use/maintainability of it.
+
+The survey received 33 responses total and free text responses have been
+paraphrased for brevity.
+
+## Introduction
+
+### Do you use GOV.UK Docker?
+
+| Answer                                | Responses | Percentage |
+|---------------------------------------|-----------|------------|
+| Yes, It's how I do all my development | 10        | 30.3%      |
+| Yes, it's a key part of my workflow   | 5         | 15.2%      |
+| Yes, occasionally                     | 9         | 27.3%      |
+| No, I don't use it                    | 9         | 27.3%      |
+
+## Questions for users of GOV.UK Docker
+
+### Does GOV.UK Docker meet your needs for running GOV.UK applications?
+
+| Answer                                | Responses | Percentage |
+|---------------------------------------|-----------|------------|
+| Absolutely                            | 9         | 37.5%      |
+| In some ways, but it's missing things | 15        | 62.5%      |
+| Not really                            | 0         | 0%         |
+
+### How could GOV.UK Docker meet more of your needs?
+
+- More data replication functionality (6 responders).
+- Not all apps are there (4 responders).
+- Easier to use (2 responders).
+- Be able to test govuk-puppet locally.
+- It would be nice to have some sample data in the databases (eg. a few
+  documents in different states).
+- Making service ports (eg, elasticsearch) exposed to the host by default
+  would be handy.
+- To setup Whitehall I needed to run the Publishing API separately or it'd
+  fail. To run Whitehall frontend I need to run Static separately or a page
+  won't render.
+
+### Are there aspects of GOV.UK Docker that cause you problems?
+
+- Issues running since Ruby upgrade (3 responders).
+- Debugging why an app isn't working locally can be difficult.
+- Setup of the databases was tricky.
+- Sometimes dependant apps fail to start up.
+- Workers aren't run by default with an app.
+- Initial app build takes time.
+- Static has problems running.
+- I often get a red failed to connect to service error message.
+- It is a little tricky to set up and usually needs a bit of effort to get to
+  work.
+- The development process could be more transparent.
+- Limited documentation for completely new starters.
+- It also seems to take up quite a lot of space for each app.
+- Troubleshooting difficult as I don't know the docker well.
+
+## Questions for people who don't use GOV.UK Docker
+
+### Why don't you use GOV.UK Docker?
+
+- Mostly run apps on local machine with `--live` flag (3 responders)
+- I haven't had the opportunity to. My VM still works.
+- I missed firebreak and don't know why we're deprecating the dev VM.
+- Every time I come to use it I can never get it running first time. Usually a
+  lot of errors and takes a lot of effort to fix them.
+- Conflicts with BYOD VPN
```

Comment:
What does BYOD stand for?

Reply:
Bring your own device 😄

Reply:
I've amended this. Thanks for the explanation @issyl0, in a meta move I've linked this doc to this PR to explain that.